Framework/DataObjects/src/ScanningWorkspaceBuilder.cpp

```diff
@@ -243,30 +235,31 @@ void ScanningWorkspaceBuilder::buildPositions(
   }
 }
 
-void ScanningWorkspaceBuilder::buildInstrumentAngles(
+void ScanningWorkspaceBuilder::buildRelativeRotationsForScans(
     DetectorInfo &outputDetectorInfo) const {
   for (size_t i = 0; i < outputDetectorInfo.size(); ++i) {
     for (size_t j = 0; j < outputDetectorInfo.scanCount(i); ++j) {
```

Comment:
This is what I referred to on Slack (a bit cryptically maybe): If you would apply these rotations before merging the `DetectorInfos`, you could simply rotate a parent of the detectors via `DetectorInfo::setRoation(IComponent *, ...)`. That should also be much faster, since it uses `Eigen` directly internally. In principle rotating the full instrument would work, but that would also rotate monitors etc., so you'd have to figure out how to handle those.